packages/flame/lib/src/components/core/component.dart

```diff
@@ -727,7 +740,8 @@ class Component {
   void processPendingLifecycleEvents() {
     if (_lifecycleManager != null) {
       _lifecycleManager!.processQueues();
-      if (!_lifecycleManager!.hasPendingEvents) {
+      if (!_lifecycleManager!.hasPendingEvents &&
+          _lifecycleManager!._removedCompleter == null) {
```

Comment:
This check can't be in `hasPendingEvents` since the component will never mount if that is the case right?

Reply:
I suppose you can do something like this: 

``` 
 bool get hasPendingEvents {
    return (_children.isNotEmpty ||
            _removals.isNotEmpty ||
            _adoption.isNotEmpty ||
            _mountCompleter != null ||
            _loadCompleter != null) &&
        _removedCompleter == null;
  }
```

Reply:
Or no that wouldn't work for other things utilizing `hasPendingEvents` I think.